File: services/Dumble.SharedKernel/Dumble.SharedKernel/Enums/UserType.cs
namespace Dumble.SharedKernel.Enums;

/// <summary>
/// Account roles. Mirrors the Auth service's enum names so JWT userType claims
/// parse cleanly. New users always start as <see cref="Participant"/>; any other
/// role requires admin approval (cert verification for Trainer, business-license
/// verification for GymOwner, etc.).
///
/// <para><see cref="GymOwner"/> is the human person who owns one or more gyms.</para>
/// <para><see cref="Gym"/> is the gym page/account itself, separate from its owner.</para>
/// </summary>
public enum UserType
{
    Participant,
    Moderator,
    Trainer,
    GymOwner,
    Gym,
    Admin
}

File: services/Dumble.Authentication/src/main/java/com/example/DumbleAuthentication/domain/UserType.java
package com.example.DumbleAuthentication.domain;

/**
 * Account roles. Every new user starts as PARTICIPANT (see AuthService.register).
 * Anything else requires admin approval (e.g. cert verification for TRAINER,
 * business-license verification for GYM_OWNER).
 *
 * GYM_OWNER is the human person who owns one or more gyms.
 * GYM is the gym page/account itself, separate from its owner.
 */
public enum UserType {
    PARTICIPANT,
    MODERATOR,
    TRAINER,
    GYM_OWNER,
    GYM,
    ADMIN
}
